### DISCUSSION TITLE
Viewer not authorized to delete even when using GITHUB_TOKEN?

### DISCUSSION THREAD

**Discussion by dlondero** [community/community#24302] [Apps, API and Webhooks]
<p>Hi there, I’m developing a Javascript action that reads a list of issues from graphql api (works!) and then deletes them based on some rules (doesn’t work yet!).</p>
<p>Looks like the error happens when using <code>GITHUB_TOKEN</code> (from <code>${{ secrets.GITHUB_TOKEN }}</code>) even if in the <code>Set Up Job</code> of the action I can see that permissions for issues look fine <code>Issues: write</code>. The call to deleteIssue fails an I get the error <code>Viewer not authorized to delete</code>.</p>
<p>Instead if I set a repository secret using a <strong>personal token</strong> (created with permissions on repo) the deletion <strong>works as expected</strong>.</p>
<p>So the question is: is the <code>GITHUB_TOKEN</code> having different permissions than the ones displayed during the set up of the action execution?</p>

**Comment by byrneh** [comment]
<p>Hi <a class="mention" href="/dlondero">@dlondero</a>, welcome to the community</p>
<p>Just a couple of quick thoughts that may or may not be a constraint depending on your actual scenario.</p>
<p>Is the issue you are trying to delete, in the same repository that contains your workflow. I read that The GITHUB_TOKEN secret permissions are limited to the repository that contains your workflow, if it is not you will need to go down the personal access token route you have already proven.</p>
<p>Or is this a personal or organization repository, as by default issues cannot be deleted in an organization’s repositories. An organization owner must enable this feature for all of the organization’s repositories first.</p>

**Comment by dlondero** [comment]
<p>Hi <a class="mention" href="/byrneh">@byrneh</a>, thanks for your help. Good questions.</p>
<p>I’m trying to delete issues on the same repository where the workflow is running. This is an organisation repository and the configuration to <code>Allow members to delete issues for this organisation</code> is already enabled at org level.</p>
<p>That’s why I find it weird for the GITHUB_TOKEN to not work as expected even though the write permission appears to be available for the issues.</p>

**Comment by dlondero** [comment, marked as answer]
<p>Ok so reading here <a href="https://docs.github.com/en/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization" class="inline-onebox" rel="noopener nofollow ugc">Repository permission levels for an organization - GitHub Docs</a> it looks like what is happening is the expected behaviour because <code>Delete an issue</code> actually requires <code>Admin</code> permissions which are not granted to the <code>GITHUB_TOKEN</code>.</p>
<p>So I guess going with the personal access token is the only option for this.</p>

**Comment by byrneh** [comment]
<p>Yes I suspect that is the case, so working as designed from a GitHub standpoint.</p>
<p>You have probably already seen, but if not, a summary in docs is here: <a href="https://docs.github.com/en/actions/reference/authentication-in-a-workflow#permissions-for-the-github_token" rel="noopener nofollow ugc">permissions-for-the-github_token</a></p>

**Comment by dlondero** [comment]
<p>Yep, saw that thanks!</p>
